src/main/scala/cromwell/engine/backend/local/LocalEngineFunctions.scala

```diff
@@ -64,7 +64,19 @@ class LocalEngineFunctions(cwd: Path, stdout: Path, stderr: Path) extends WdlSta
    * Try to read an integer from the file referenced by the specified `WdlValue`.
    */
   override protected def read_int(params: Seq[Try[WdlValue]]): Try[WdlInteger] =
-    read_string(params).map { s => WdlInteger(s.value.trim.toInt) }
+    read_string(params) map { s => WdlInteger(s.value.trim.toInt) }
+
+  /**
+   * Try to read a float from the file referenced by the specified `WdlValue`.
+   */
+  override protected def read_float(params: Seq[Try[WdlValue]]): Try[WdlFloat] =
+    read_string(params) map { s => WdlFloat(s.value.trim.toDouble) }
+
+  /**
+   * Try to read a boolean from the file referenced by the specified `WdlValue`.
+   */
+  override protected def read_boolean(params: Seq[Try[WdlValue]]): Try[WdlBoolean] =
+    read_string(params) map { s => WdlBoolean(java.lang.Boolean.parseBoolean(s.value.trim.toLowerCase)) }
```

Comment:
No.  Actually I should refactor read_int/read_float/read_boolean up to the trait.